Source/Core/DolphinWX/LuaScriptFrame.cpp

```diff
@@ -0,0 +1,553 @@
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getAnalog", getAnalog));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setAnalog", setAnalog));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getCStick", getCStick));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setCStick", setCStick));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setCStickPolar", setCStickPolar));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setAnalogPolar", setAnalogPolar));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getButtons", getButtons));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setButtons", setButtons));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setDPad", setDPad));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getTriggers", getTriggers));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setTriggers", setTriggers));
+  }
+}
+
+LuaScriptFrame::~LuaScriptFrame()
+{
+  // Disconnect Events
+  this->Disconnect(wxID_ANY, wxEVT_COMMAND_MENU_SELECTED,
+                   wxCommandEventHandler(LuaScriptFrame::OnClearClicked));
+  browse_button->Disconnect(wxEVT_COMMAND_BUTTON_CLICKED,
+                            wxCommandEventHandler(LuaScriptFrame::BrowseOnButtonClick), NULL, this);
+  run_button->Disconnect(wxEVT_COMMAND_BUTTON_CLICKED,
+                         wxCommandEventHandler(LuaScriptFrame::RunOnButtonClick), NULL, this);
+  stop_button->Disconnect(wxEVT_COMMAND_BUTTON_CLICKED,
+                          wxCommandEventHandler(LuaScriptFrame::StopOnButtonClick), NULL, this);
+
+  // Stop currently executing Lua thread
+  if (lua_thread != nullptr && lua_thread->IsRunning())
+  {
+    lua_thread->Kill();
+    lua_thread = nullptr;
+  }
+
+  main_frame->m_lua_script_frame = nullptr;
+
+  // Nullify GC manipulator function to prevent crash when lua console is closed
+  Movie::s_gc_manip_funcs[Movie::GCManipIndex::LuaGCManip] = nullptr;
+}
+
+//
+// CreateGUI
+//
+// Creates actual Lua console window.
+//
+void LuaScriptFrame::CreateGUI()
+{
+  this->SetSizeHints(wxDefaultSize, wxDefaultSize);
+
+  m_menubar = new wxMenuBar(0);
+  console_menu = new wxMenu();
+  clear =
+      new wxMenuItem(console_menu, wxID_ANY, wxString(_("Clear")), wxEmptyString, wxITEM_NORMAL);
+  console_menu->Append(clear);
+  help_menu = new wxMenu();
+  documentation = new wxMenuItem(help_menu, wxID_ANY, wxString(_("Lua Documentation")),
+                                 wxEmptyString, wxITEM_NORMAL);
+  api = new wxMenuItem(help_menu, wxID_ANY, wxString(_("Dolphin Lua API")), wxEmptyString,
+                       wxITEM_NORMAL);
+  help_menu->Append(documentation);
+  help_menu->Append(api);
+
+  m_menubar->Append(console_menu, _("Console"));
+  m_menubar->Append(help_menu, _("Help"));

```

Comment:
This and the above relevant code should likely be put in its own `CreateMenuBar()` function:

in such a way its possible to just do:

```cpp
SetMenuBar(CreateMenuBar());
```